src/lib.rs

```diff
@@ -198,14 +218,68 @@ impl Expr {
         }
     }
 
-    pub fn as_json(&self) -> String {
-        serde_json::to_string(&self).unwrap()
+    /// Converts this expression to a JSON string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cql2::Expr;
+    ///
+    /// let expr = Expr::Bool(true);
+    /// let s = expr.as_json().unwrap();
+    /// ```
+    pub fn as_json(&self) -> Result<String, serde_json::Error> {
+        serde_json::to_string(&self)
+    }
+
+    /// Converts this expression to a pretty JSON string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cql2::Expr;
+    ///
+    /// let expr = Expr::Bool(true);
+    /// let s = expr.as_json_pretty().unwrap();
+    /// ```
+    pub fn as_json_pretty(&self) -> Result<String, serde_json::Error> {
+        serde_json::to_string_pretty(&self)
     }
-    pub fn as_json_pretty(&self) -> String {
-        serde_json::to_string_pretty(&self).unwrap()
+
+    /// Converts this expression to a [serde_json::Value].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cql2::Expr;
+    ///
+    /// let expr = Expr::Bool(true);
+    /// let value = expr.to_value().unwrap();
+    /// ```
+    pub fn to_value(&self) -> Result<Value, serde_json::Error> {
+        serde_json::to_value(self)
     }
-    pub fn validate(&self) -> bool {
-        Validator::new().validate_str(&self.as_json())
+
+    /// Returns true if this expression is valid CQL2.
+    ///
+    /// For detailed error reporting, use [Validator::validate] in conjunction with [Expr::to_value].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cql2::Expr;
+    ///
+    /// let expr = Expr::Bool(true);
+    /// assert!(expr.is_valid());
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// Panics if the default validator can't be created.
+    pub fn is_valid(&self) -> bool {
+        let value = serde_json::to_value(self).unwrap();
+        let validator = Validator::new().unwrap();
+        validator.validate(&value).is_ok()
```

Comment:
I think we can be a bit more conservative here, and consider a non-json-able expression (which probably shouldn't exist) as invalid:

```suggestion
        let value = serde_json::to_value(self).map(|value| {
            let validator = Validator::new().expect("should be able to create a default validator");
            validator.validate(&value).is_ok()
        }).unwrap_or_default();
```

NB this suggestion will fail the linter/formatter, probably.